lib/multi_op_queue.rb

```diff
@@ -132,7 +140,8 @@ def pop_up_to(num_to_pop = 1, non_block=false)
               else
                 begin
                   @num_waiting += 1
-                  @cond.wait @mutex
+                  @cond.wait(@mutex, timeout)
+                  return nil if @que.empty?
```

Comment:
what's the thinking behind `nil` vs `[]` ? `pop_up_to` was build to always return an enumerable

Reply:
i was mimicking other APIs that have a timeout, e.g., `select`.

Reply:
I guess in this case a `[]` is basically the same as `nil` since before a block would only yield with results. I'm happy either way, though.

Reply:
well, the interface also breaks either way since `non_block` is changing to a hash argument